samples/extensions/open_cl_interop/open_cl_interop.cpp

```diff
@@ -0,0 +1,622 @@
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	cl_data = new CLData{};
+
+	prepare_open_cl_resources();
+	prepare_shared_resources();
+	generate_quad();
+	prepare_uniform_buffers();
+	setup_descriptor_set_layout();
+	prepare_pipelines();
+	setup_descriptor_pool();
+	setup_descriptor_set();
+	build_command_buffers();
+
+	auto fence_create_info = vkb::initializers::fence_create_info(VK_FENCE_CREATE_SIGNALED_BIT);
+	vkCreateFence(device->get_handle(), &fence_create_info, nullptr, &rendering_finished_fence);
+
+	prepared = true;
+	return true;
+}
+
+void OpenCLInterop::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
+	total_time_passed += delta_time;
+
+	// Wait until Vulkan rendering is finished and the texture can be written to
+	vkWaitForFences(device->get_handle(), 1, &rendering_finished_fence, VK_TRUE, std::numeric_limits<uint64_t>::max());
+	vkResetFences(device->get_handle(), 1, &rendering_finished_fence);
+
+	// Fill the texture using OpenCL
+	run_texture_generation();
+
+	// Wait until the texture is filled
+	// Using synchronization primitives would be better, but corresponding OpenCL extensions are not yet available
+	clFlush(cl_data->command_queue);
+	clFinish(cl_data->command_queue);
+
+	// Display the texture using Vulkan
+	ApiVulkanSample::prepare_frame();
+
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, rendering_finished_fence));
+
+	ApiVulkanSample::submit_frame();
+}
+
+void OpenCLInterop::view_changed()
+{
+	update_uniform_buffers();
+}
+
+void OpenCLInterop::build_command_buffers()
+{
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	VkClearValue clear_values[2];
+	clear_values[0].color        = default_clear_color;
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderPass               = render_pass;
+	render_pass_begin_info.renderArea.offset.x      = 0;
+	render_pass_begin_info.renderArea.offset.y      = 0;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = 2;
+	render_pass_begin_info.pClearValues             = clear_values;
+
+	for (int32_t i = 0; i < draw_cmd_buffers.size(); i++)
+	{
+		render_pass_begin_info.framebuffer = framebuffers[i];
+
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+
+		vkCmdBeginRenderPass(draw_cmd_buffers[i], &render_pass_begin_info, VK_SUBPASS_CONTENTS_INLINE);
+
+		VkViewport viewport = vkb::initializers::viewport((float) width, (float) height, 0.0f, 1.0f);
+		vkCmdSetViewport(draw_cmd_buffers[i], 0, 1, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);

```

Comment:
width/height are uint32_t static_cast them from the int32_t type they are here for warning removal.